src/main/kotlin/com/atlassian/performance/tools/report/api/judge/Verdict.kt

```diff
@@ -14,20 +14,42 @@ class Verdict internal constructor(
     val reports: List<JUnitReport>,
     val failedActions: List<ActionType<*>>
 ) {
+    private companion object {
+        private val logger = LogManager.getLogger(this::class.java)
+    }
 
+    @Deprecated("Use Builder")
     constructor(
         reports: List<JUnitReport>
     ) : this(
         reports = reports,
         failedActions = emptyList()
     )
 
-    private val logger = LogManager.getLogger(this::class.java)
+    /**
+     * Since 3.12.0
+     */
+    class Builder(private val reports: List<JUnitReport>) {
+        private var failedActions: List<ActionType<*>> = emptyList()
+
+        fun failedActions(failedActions: List<ActionType<*>>): Builder {
+            this.failedActions = failedActions
+            return this
+        }
+
+        fun build(): Verdict {
+            return Verdict(
+                reports = reports,
+                failedActions = failedActions
+            )
+        }
+    }
+
 
     /**
      * @since 3.9.0
      */
-    val positive: Boolean = reports.all { it.successful }
+    val positive: Boolean by lazy { reports.all { it.successful } }
```

Comment:
Doesn't hurt, but why?

Reply:
I thought the opposite while doing this :D "it doesn't hurt, but why not?"
Yeah, I'll remove it. Brings no value to the table

Reply:
Note: If we would actually need it I think that caching could be made better when implementing it explicitly using a composite, e.g. `CachingVerdict`. This would require and interface or open class though.